### ISSUE TITLE
 64-bit private key not found

### ISSUE THREAD

**Issue by filo1992** [brichard19/BitCrack#257]
I have finished the 64 bit range, but I have not found any key with 8 rx 570 8gb.

**Comment by RastaDollo** [comment]
Very interesting
Repeat in https://bitcointalk.org/index.php?topic=4453897.0
What says in forum

**Comment by MRGiacalone** [comment]
> I have finished the 64 bit range, but I have not found any key with 8 rx 570 8gb.

Aprox. how long did it take you with those 8 GPU's?

**Comment by filo1992** [comment]
> > Ho terminato l'intervallo a 64 bit, ma non ho trovato alcuna chiave con 8 rx 570 8 gb.
> 
> Aprox. quanto tempo hai impiegato con quelle 8 GPU?


about 15 days

**Comment by RB61** [comment]
> I have finished the 64 bit range, but I have not found any key with 8 rx 570 8gb.

with 8 rx570 you can check let's say around 10^9 case/s.
puzzle 64 requires  2^63 cases which is around 10^19 cases. So you need to wait 10^10 seconds to claim spaning the whole range. 
10^10 sec is over 300 years.
have you fixed the opencl version according to
https://github.com/brichard19/BitCrack/issues/256

**Comment by filo1992** [comment]
> > Ho terminato l'intervallo a 64 bit, ma non ho trovato alcuna chiave con 8 rx 570 8 gb.
> 
> con 8 rx570 puoi controllare diciamo circa 10 ^ 9 caso / i.
> il puzzle 64 richiede 2 ^ 63 casi, ovvero circa 10 ^ 19 casi. Quindi è necessario attendere 10 ^ 10 secondi per richiedere di estendere l'intero intervallo.
> 10 ^ 10 sec è di oltre 300 anni.
> hai corretto la versione opencl secondo
> [# 256](https://github.com/brichard19/BitCrack/issues/256)

with an rx 570 I can get 160Mk / s.

I had tested the first 23 private keys, and had found no errors. now I try to make the change that you recommend

**Comment by RB61** [comment]
> > > Ho terminato l'intervallo a 64 bit, ma non ho trovato alcuna chiave con 8 rx 570 8 gb.
> > 
> > 
> > con 8 rx570 puoi controllare diciamo circa 10 ^ 9 caso / i.
> > il puzzle 64 richiede 2 ^ 63 casi, ovvero circa 10 ^ 19 casi. Quindi è necessario attendere 10 ^ 10 secondi per richiedere di estendere l'intero intervallo.
> > 10 ^ 10 sec è di oltre 300 anni.
> > hai corretto la versione opencl secondo
> > [# 256](https://github.com/brichard19/BitCrack/issues/256)
> 
> with an rx 570 I can get 160Mk / s.
> 
> I had tested the first 23 private keys, and had found no errors. now I try to make the change that you recommend

160Mk per  card seems way to much.
what parameters -b -t -p do you use and what is your core clock and memory clock ?
my Rx480 8Gb has 36 computing units, cclock =1266 mclock=2000
I use -b 72 -t 256 -p 1024  and I get 126 Mk/s

**Comment by RB61** [comment]
> > > Ho terminato l'intervallo a 64 bit, ma non ho trovato alcuna chiave con 8 rx 570 8 gb.
> > 
> > 
> > con 8 rx570 puoi controllare diciamo circa 10 ^ 9 caso / i.
> > il puzzle 64 richiede 2 ^ 63 casi, ovvero circa 10 ^ 19 casi. Quindi è necessario attendere 10 ^ 10 secondi per richiedere di estendere l'intero intervallo.
> > 10 ^ 10 sec è di oltre 300 anni.
> > hai corretto la versione opencl secondo
> > [# 256](https://github.com/brichard19/BitCrack/issues/256)
> 
> with an rx 570 I can get 160Mk / s.
> 
> I had tested the first 23 private keys, and had found no errors. now I try to make the change that you recommend

160Mk per  card seems way to much.
what parameters -b -t -p do you use and what is your core clock and memory clock ?
my Rx480 8Gb has 36 computing units, cclock =1266 mclock=2000
I use -b 72 -t 256 -p 1024  and I get 126 Mk/s

**Comment by filo1992** [comment]
> > > > Ho terminato l'intervallo a 64 bit, ma non ho trovato alcuna chiave con 8 rx 570 8 gb.
> > > 
> > > 
> > > con 8 rx570 puoi controllare diciamo circa 10 ^ 9 caso / i.
> > > il puzzle 64 richiede 2 ^ 63 casi, ovvero circa 10 ^ 19 casi. Quindi è necessario attendere 10 ^ 10 secondi per richiedere di estendere l'intero intervallo.
> > > 10 ^ 10 sec è di oltre 300 anni.
> > > hai corretto la versione opencl secondo
> > > [# 256](https://github.com/brichard19/BitCrack/issues/256)
> > 
> > 
> > with an rx 570 I can get 160Mk / s.
> > I had tested the first 23 private keys, and had found no errors. now I try to make the change that you recommend
> 
> 160Mk per card seems way to much.
> what parameters -b -t -p do you use and what is your core clock and memory clock ?
> my Rx480 8Gb has 36 computing units, cclock =1266 mclock=2000
> I use -b 72 -t 256 -p 1024 and I get 126 Mk/s


use -b 32 -t 256 -p 2000, cclock = 1175 mclock = 2090 and I get 158 Mk/s

**Comment by RB61** [comment]
Excellent card :) !  mine is probably older and cheaper lol.
have you tried -b 64 -t 256 -p 2048 ? you might get a little more 

Have you recompiled the code?

**Comment by filo1992** [comment]
> Ottima carta :)! il mio è probabilmente più vecchio ed economico lol.
> hai provato -b 64 -t 256 -p 2048? potresti ottenere un po 'di più
> 
> Hai ricompilato il codice?


i am having some problems in compiling with vs.

yes I tried parameters, but the pc freezes (SYSTEM_THREAD_EXCEPTION_NOT_HANDLED atikmdag.sys).

**Comment by RB61** [comment]
So maybe the max for your card  -b is 32 when you use -t 256 
but 158 Mk/s is excellent already.
So 8 x 158Mk/s  = 1264 Mk/s = 1 355 400 064 k/s 
2^63 = 9 223 372 036 854 775 808
9 223 372 036 854 775 808 / 1 355 400 064 = 6 958 934 353 seconds = 80 543 days = 220 years

**Comment by filo1992** [comment]
> Quindi forse il massimo per la tua scheda -b è 32 quando usi -t 256
> ma 158 Mk / s è già eccellente.
> Quindi 8 x 158Mk / s = 1264 Mk / s = 1 355 400 064 k / s
> 2 ^ 63 = 9 223 372 036 854 775 808
> 9 223 372 036 854 775 808/1355 400 064 = 6 958 934 353 secondi = 80 543 giorni = 220 anni


yes it seemed very strange to take so little time, given the large range.

but using a client / server version of Etar, dividing the spaces by 2 ^ 12. the program said range finished

**Comment by RB61** [comment]
I did know there was a client/server version?  I asked bitchard19 about it, he said he was working on it:
https://bitcointalk.org/index.php?topic=5244940.msg54671227#msg54671227


do you have github link to Etar ?

**Comment by filo1992** [comment]
> I did know there was a client/server version? I asked bitchard19 about it, he said he was working on it:
> https://bitcointalk.org/index.php?topic=5244940.msg54671227#msg54671227
> 
> do you have github link to Etar ?


i use this https://bitcointalk.org/index.php?topic=5244940.msg54666524#msg54666524.


I'm also waiting for the original version of bitchard19

**Comment by RB61** [comment]
You should not use .exe from anyone, could be malware ... I read Etar's post , he is claiming impossible things lol
how can a CPU with GHz scale clocks can do TH/s lol it is a joke lol

**Comment by MRGiacalone** [comment]
> > Quindi forse il massimo per la tua scheda -b è 32 quando usi -t 256
> > ma 158 Mk / s è già eccellente.
> > Quindi 8 x 158Mk / s = 1264 Mk / s = 1 355 400 064 k / s
> > 2 ^ 63 = 9 223 372 036 854 775 808
> > 9 223 372 036 854 775 808/1355 400 064 = 6 958 934 353 secondi = 80 543 giorni = 220 anni
> 
> yes it seemed very strange to take so little time, given the large range.
> 
> but using a client / server version of Etar, dividing the spaces by 2 ^ 12. the program said range finished

9 223 372 036 854 775 808/1355 400 064
Is a 64 bit Decimal number.

The key space is a 256-bit Decimal number, or a 64 digit (characters) Hexadecimal. Up to:
FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFE BAAE DCE6 AF48 A03B BFD2 5E8C D036 4140

**Comment by filo1992** [comment]
> Non dovresti usare .exe da nessuno, potrebbe essere un malware ... Ho letto il post di Etar, sta sostenendo cose impossibili lol
> come può una CPU con clock in scala GHz fare TH / s lol è uno scherzo lol

yes I read about that post, and as I understood it was correct and apologized.

in the exe files there are also the .pb files that you can check and compile.

**Comment by RB61** [comment]
> > > Quindi forse il massimo per la tua scheda -b è 32 quando usi -t 256
> > > ma 158 Mk / s è già eccellente.
> > > Quindi 8 x 158Mk / s = 1264 Mk / s = 1 355 400 064 k / s
> > > 2 ^ 63 = 9 223 372 036 854 775 808
> > > 9 223 372 036 854 775 808/1355 400 064 = 6 958 934 353 secondi = 80 543 giorni = 220 anni
> > 
> > 
> > yes it seemed very strange to take so little time, given the large range.
> > but using a client / server version of Etar, dividing the spaces by 2 ^ 12. the program said range finished
> 
> 9 223 372 036 854 775 808/1355 400 064
> Is a 64 bit Decimal number.
> 
> The key space is a 256-bit Decimal number, or a 64 digit (characters) Hexadecimal. Up to:
> FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFE BAAE DCE6 AF48 A03B BFD2 5E8C D036 4140

puzzle 64bit is from 8000 0000 0000 0000 to FFFF FFFF FFFF FFFF which is 2^63 = 9223372036854775808 in decimal
if you divide your rate which is 1355400064 keys/s  you get 6 958 934 353 secondes which is 220 years.

**Comment by RB61** [comment]
> > Non dovresti usare .exe da nessuno, potrebbe essere un malware ... Ho letto il post di Etar, sta sostenendo cose impossibili lol
> > come può una CPU con clock in scala GHz fare TH / s lol è uno scherzo lol
> 
> yes I read about that post, and as I understood it was correct and apologized.
> 
> in the exe files there are also the .pb files that you can check and compile.

.pb files ? where ? 
Anyhow, he uses windows, I use Linux ...

**Comment by filo1992** [comment]
> > > Non dovresti usare .exe da nessuno, potrebbe essere un malware ... Ho letto il post di Etar, sta sostenendo cose impossibili lol
> > > come può una CPU con clock in scala GHz fare TH / s lol è uno scherzo lol
> > 
> > 
> > yes I read about that post, and as I understood it was correct and apologized.
> > in the exe files there are also the .pb files that you can check and compile.
> 
> .pb files ? where ?
> Anyhow, he uses windows, I use Linux ...

sorry I was wrong to share the link 
https://drive.google.com/file/d/1pFTvBLwTDF4GZCyDpJHwnWqfuNeOT6Ik/edit

**Comment by RB61** [comment]
IncludeFile "E:\Program Files\PureBasic\Examples\Curve64.pb" is missing

**Comment by RB61** [comment]
pb is purebasic right?
I am installing purebasic now and I will try to compile.

**Comment by filo1992** [comment]
> pb è purebasic giusto?
> Sto installando purebasic ora e proverò a compilare.

I think so

**Comment by RB61** [comment]
I let you know what I get ...

**Comment by filo1992** [comment]
> Ti faccio sapere cosa ottengo ...

Thanks

**Comment by RB61** [comment]
I have checked. These pb programs are scripts that use already compiled  clBitcrack or cuBitcrack .... 

just recompile your clBitCrack with changes as  I mentioned here : 

https://github.com/brichard19/BitCrack/issues/256

**Comment by filo1992** [comment]
I can't compile vs I find this error 
Severity Code Description Project File Row Elimination status
MSB6006 error "CL.exe" ended with code 2. clUtil C: \ Program Files (x86) \ Microsoft Visual Studio \ 2019 \ Community \ MSBuild \ Microsoft \ VC \ v160 \ Microsoft.CppCommon.targets 514

**Comment by RB61** [comment]
Sorry, I use linux. Maybe you should ask Brichard19 about this error.

**Comment by Vovkaruss** [comment]
a hint for those looking for the key to puzzle number 64! Use the normal distribution property for statistical analysis of ranges. Satoshi used uniform distribution - so if you search in those points 1 ... 100 where there were no hits 1-63 keys before, you will increase the chances of finding a key 63 times - up to about 3% per month using a speed of about 500MH/s

**Comment by dlasher** [comment]
Same problem, entire keyspace searched, no match for : 16jY7qLJnxb7CHZyqBP8qca9d51gAjyXQN

64 | 263...264 | 8000000000000000...ffffffffffffffff

Broke into two chunks, one per GPU, Invoked as :

bin/clBitCrack --keyspace 7fffffffffffffff:c000000000000000 -d 1 16jY7qLJnxb7CHZyqBP8qca9d51gAjyXQN -b 64 -t 512 -p 1024
[2020-10-09.07:41:21] [Info] Starting at: 0000000000000000000000000000000000000000000000007FFFFFFFFFFFFFFF
[2020-10-09.07:41:21] [Info] Ending at:   000000000000000000000000000000000000000000000000C000000000000000
[2020-10-09.07:41:21] [Info] Counting by: 0000000000000000000000000000000000000000000000000000000000000001
GeForce GTX 970  2304 / 4043MB | 1 target 73.84 MKey/s (140,737,404,469,248 total) [22:01:51:32][2020-10-31.09:33:43] [Info] Reached end of keyspace

bin/clBitCrack --keyspace c000000000000000:ffffffffffffffff -d 2 16jY7qLJnxb7CHZyqBP8qca9d51gAjyXQN -b 64 -t 512 -p 768
[2020-10-09.07:42:17] [Info] Starting at: 000000000000000000000000000000000000000000000000C000000000000000
[2020-10-09.07:42:17] [Info] Ending at:   000000000000000000000000000000000000000000000000FFFFFFFFFFFFFFFF
[2020-10-09.07:42:17] [Info] Counting by: 0000000000000000000000000000000000000000000000000000000000000001
GeForce GTX 970  2304 / 4041MB | 1 target 73.05 MKey/s (140,737,404,469,248 total) [22:07:31:54][2020-10-31.15:14:42] [Info] Reached end of keyspace

**Comment by Alf71** [comment]
And are you sure that the privkey is contained inside of the range between
0000000000000000000000000000000000000000000000007FFFFFFFFFFFFFFF
and
000000000000000000000000000000000000000000000000C000000000000000?

**Comment by dlasher** [comment]
Look at the 64 bit key on this list : https://privatekeys.pw/puzzles/bitcoin-puzzle-tx

Unless they, and all the other places that post this list, are wrong?

START: 0000000000000000000000000000000000000000000000007FFFFFFFFFFFFFFF
END: 000000000000000000000000000000000000000000000000FFFFFFFFFFFFFFFF

**Comment by dlasher** [comment]
Works with any of the shorter searches I try, like #6  - 1PitScNLyp2HCygzadCh7FveTnfmpPbfp8

bin/clBitCrack --keyspace 0000000000000001:00000000000000FF -d 1 1PitScNLyp2HCygzadCh7FveTnfmpPbfp8 -b 64 -t 512 -p 768              
[2020-10-31.20:29:55] [Info] Compression: compressed
[2020-10-31.20:29:55] [Info] Starting at: 0000000000000000000000000000000000000000000000000000000000000001
[2020-10-31.20:29:55] [Info] Ending at:   00000000000000000000000000000000000000000000000000000000000000FF
[2020-10-31.20:29:55] [Info] Counting by: 0000000000000000000000000000000000000000000000000000000000000001
[2020-10-31.20:29:55] [Info] Compiling OpenCL kernels...
[2020-10-31.20:29:55] [Info] Initializing GeForce GTX 970
[2020-10-31.20:30:02] [Info] Generating 25,165,824 starting points (960.0MB)
[2020-10-31.20:30:10] [Info] 10.0%
[2020-10-31.20:30:13] [Info] 20.0%
[2020-10-31.20:30:15] [Info] 30.0%
[2020-10-31.20:30:17] [Info] 40.0%
[2020-10-31.20:30:19] [Info] 50.0%
[2020-10-31.20:30:21] [Info] 60.0%
[2020-10-31.20:30:22] [Info] 70.0%
[2020-10-31.20:30:23] [Info] 80.0%
[2020-10-31.20:30:24] [Info] 90.0%
[2020-10-31.20:30:25] [Info] 100.0%
[2020-10-31.20:30:25] [Info] Done
[2020-10-31.20:30:26] [Info] Address:     1PitScNLyp2HCygzadCh7FveTnfmpPbfp8
                             Private key: 0000000000000000000000000000000000000000000000000000000000000031
                             Compressed:  yes
                             Public key:  
                             03F2DAC991CC4CE4B9EA44887E5C7C0BCE58C80074AB9D4DBAEB28531B7739F530
                             
[2020-10-31.20:30:26] [Info] No targets remaining
[2020-10-31.20:30:26] [Info] Reached end of keyspace

**Comment by dlasher** [comment]
I do see two possible causes:
- https://github.com/brichard19/BitCrack/issues/256
- https://github.com/brichard19/BitCrack/issues/263

**Comment by hamnaz** [comment]
you checked 140,737,404,469,248 + 140,737,404,469,248 total keys 281,474,808,938,496

281474808938496
and in 64 bit range total keys need to be checked
9223372036854775807
check where you have mistake, you did not complete full check

**Comment by dlasher** [comment]
> you checked 140,737,404,469,248 + 140,737,404,469,248 total keys 281,474,808,938,496
> 
> 281474808938496
> and in 64 bit range total keys need to be checked
> 9223372036854775807
> check where you have mistake, you did not complete full check

I see the math.. check the cmd-line parameters I used.. did I miss something or BitCrack miss something?

**Comment by hamnaz** [comment]
-b 64 -t 512 -p 768 maybe for cuda, not for cl, btw they mention could be bugy at amd or intel devices, for best use bitcrack is cuda
